src/responsibleai/rai_analyse/generate_pdf.py

```diff
@@ -0,0 +1,1226 @@
+            )
+        )
+        main_elems.append(
+            p(
+                "What data points experience the largest impact (causal effect) to changes"
+                ", while treating \"{}\"".format(f["feature"])
+            )
+        )
+
+        main_elems.append(
+            get_table(
+                [[p["index"], p["Current treatment"], p["Treatment"], round(p["Effect of treatment"], 2)]
+                 for p in data["top_local_policies"][f["feature"]]])
+        )
+
+    main_container = div(main_elems, _class="main")
+
+    return div(
+        div(
+            get_page_divider("Causal"),
+            left_container,
+            main_container,
+            _class="nobreak_div"
+        ), _class="container nobreak_div")
+
+
+class ClassificationComponents:
+    @staticmethod
+    def get_model_overview_page(data):
+        return get_model_overview(data)
+
+    @staticmethod
+    def get_data_explorer_page(data):
+        de_heading_left_elms = p("Evaluate your dataset to assess representation of identified subgroups:")
+
+        de_heading_left_container = div(de_heading_left_elms, _class="left")
+
+        de_main_elems = []
+
+        def get_de_bar_plot(data):
+            class_0 = data[0]["prediction"][0]
+
+            y_data = [str(c["label"]) + "<br>" + str(int(100 * c["population"])) + "%" for c in data]
+            x_data = [100 * (list(c["prediction"]).count(class_0))/(len(c["prediction"])) for c in data]
+            x_data = [[x, 100 - x] for x in x_data]
+
+            tickvals = [0, 25, 50, 75, 100]
+            ticktext = [str(x) + "%" for x in tickvals]
+            legend = ["Predicted as {}".format(class_0)]
+
+            png_base64 = get_bar_plot(y_data, x_data, legend=legend,
+                                      tickvals=tickvals, ticktext=ticktext, tickappend="%")
+            return div(img(_src="data:image/png;base64,{}".format(png_base64)), _class="image_div")
+
+        for c in data:
+            de_main_elems.append(h3(c["feature_name"]))
+            for i in c["data"]:
+                de_main_elems.append(p(
+                    "\"{}\" have {}% {}".format(
+                        i["label"], round(i[c["primary_metric"]]*100, 1), c["primary_metric"]
+                    )))
+            de_main_elems.append(
+                div(
+                    p("Predicted classification output of the different subgroups are as follows:"),
+                    get_de_bar_plot(c["data"]),
+                    _class="nobreak_div"
+                )
+            )
+
+        de_main_container = div(de_main_elems, _class="main")
+
+        return str(
+            div(get_page_divider("Data Explorer"), de_heading_left_container, de_main_container, _class="container"))
+
+    @staticmethod
+    def _get_model_performance_explanation_text(metric, data):
+        if metric == "accuracy_score":
+            return div(
```

Comment:
Personally, I'm not a huge fan of gigantic statements like this. Especially not in `return`.

Reply:
to be fixed in follow up.